Microsoft.Kiota.Http.HttpClientLibrary.Tests/KiotaClientFactoryTests.cs

```diff
@@ -87,6 +89,31 @@ public void GetDefaultHttpMessageHandlerSetsUpProxy()
 
         }
 
+        [Fact]
+        public void CreateDefaultHandlersWithOptions()
+        {
+            // Arrange
+            var retryHandlerOption = new RetryHandlerOption { MaxRetry = 5, ShouldRetry = (_, _, _) => true };
+
+
+            // Act
+            var handlers = KiotaClientFactory.CreateDefaultHandlers(new[] { retryHandlerOption });
+
+            RetryHandler retryHandler = default;
+            foreach(var handler in handlers)
+            {
+                if(handler is RetryHandler retryFromDefault)
+                {
+                    retryHandler = retryFromDefault;
+                    break;
+                }
+            }
+
+            // Assert
+            Assert.NotNull(retryHandler);
+            Assert.NotNull(retryHandler.RetryOption);

```

Comment:
the option is initialized by default, here I'd rely on a reference equal with the reference you have in the arrange section instead.